BBKoffieTuin/Assets/Scripts/GpsService.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Android;
+using UnityEngine.Events;
+
+public class GpsService : MonoBehaviour
+{
+    [SerializeField] private int maxWaitInSeconds = 15;
+
+    public UnityEvent onLocationServicesStarted = new UnityEvent();
+    public bool GpsServiceEnabled { get; private set; } = false;
+
+    private bool _hasFineLocationPermission = false;
+    private PermissionCallbacks _permissionCallbacks;
+    
+    IEnumerator Start()
+    {
+        //handle permission
+        _hasFineLocationPermission = Permission.HasUserAuthorizedPermission(Permission.FineLocation);
+        
+        //WE HAVE PERMISSION SO WE CAN START THE SERVICE
+        if (_hasFineLocationPermission)
+        {
+            StartCoroutine(StartLocationServices());
+            yield break;
+        }
+
+        _permissionCallbacks = new PermissionCallbacks();
+        //WE DON'T HAVE PERMISSION SO WE REQUEST IT AND START SERVICES ON GRANTED.
+        _permissionCallbacks = new PermissionCallbacks();
```

Comment:
2x?

Reply:
waarschijnlijk omdat jij het ook hebt toegoevoegd komt het er nu 2x in